src/internal/list.js

```diff
@@ -0,0 +1,22 @@
+function List(head, tail){
+  this.head = head;
+  this.tail = tail;
+  this.size = tail.size + 1;
+}
+
+List.prototype.isEmpty = false;
+
+List.prototype.add = function List$add(x){
+  return new List(x, this);
+};
+
+function Empty(){}
+
+Empty.prototype = Object.create(List.prototype);
+
+Empty.prototype.isEmpty = true;
+Empty.prototype.head = null;
+Empty.prototype.tail = null;
+Empty.prototype.size = 0;
+
+export default new Empty;
```

Comment:
It's because I'm implementing the bare minimum, so to construct a list one does `list.add(1).add(2).add(3)`. No need to export the internals, or a specialized constructor.